apps/etterlatte-saksbehandling-ui/client/src/components/behandling/beregningsgrunnlag/flereAvdoede/BeregningsMetodeRadForAvdoed.tsx

```diff
@@ -0,0 +1,151 @@
+}
+
+export const BeregningsMetodeRadForAvdoed = ({
+  behandlingId,
+  ident,
+  beregningsMetodeForAvdoed,
+  navn,
+  redigerbar,
+  patchGrunnlagOppdaterMetode,
+  patchGrunnlagSlettMetode,
+}: Props) => {
+  const dispatch = useAppDispatch()
+  const [redigerModus, setRedigerModus] = useState<boolean>(false)
+  const [lagreBeregningsgrunnlagResult, lagreBeregningsgrunnlagRequest] = useApiCall(lagreBeregningsGrunnlag)
+
+  return (
+    <Table.ExpandableRow
+      open={redigerModus}
+      onOpenChange={(open) => {
+        setRedigerModus(open)
+      }}
+      key={ident}
+      content={
+        redigerModus ? (
+          <BeregningsMetodeForAvdoded
+            ident={ident}
+            navn={navn}
+            eksisterendeMetode={beregningsMetodeForAvdoed}
+            paaAvbryt={() => {
+              setRedigerModus(false)
+            }}
+            oppdaterBeregningsMetodeForAvdoed={oppdaterBeregningsMetodeForAvdoed}
+            lagreBeregningsgrunnlagResult={lagreBeregningsgrunnlagResult}
+          />
+        ) : (
+          ''
+        )
+      }
+    >
+      <Table.DataCell>{navn}</Table.DataCell>
+      <Table.DataCell>
+        {beregningsMetodeForAvdoed
+          ? beregningsMetodeForAvdoed.data.beregningsMetode.beregningsMetode
+          : 'Metode er ikke satt'}
+      </Table.DataCell>
+      <Table.DataCell>
+        {beregningsMetodeForAvdoed?.fom ? format(startOfMonth(beregningsMetodeForAvdoed.fom), 'yyyy-MM-dd') : '-'}
+      </Table.DataCell>
+      <Table.DataCell>
+        {beregningsMetodeForAvdoed?.tom ? format(startOfMonth(beregningsMetodeForAvdoed.tom), 'yyyy-MM-dd') : '-'}
+      </Table.DataCell>
+      {redigerbar &&
+        (beregningsMetodeForAvdoed ? (
+          <>
+            <Table.DataCell>{redigerKnapp()}</Table.DataCell>
+            <Table.DataCell>{slettKnapp()}</Table.DataCell>
+          </>
+        ) : (
+          <Table.DataCell>{leggTilKnapp()}</Table.DataCell>
+        ))}
+    </Table.ExpandableRow>
+  )
+
+  function redigerKnapp() {
+    return (
+      <Button
+        type="button"
+        variant="secondary"
+        size="small"
+        icon={<PencilIcon aria-hidden />}
+        disabled={redigerModus}
+        onClick={() => setRedigerModus(true)}
+      >
+        Rediger
+      </Button>
+    )
+  }
+
+  function leggTilKnapp() {
+    return (
+      <Button
+        type="button"
+        variant="secondary"
+        size="small"
+        icon={<PencilIcon aria-hidden />}
+        disabled={redigerModus}
+        onClick={() => setRedigerModus(true)}
+      >
+        Legg til
+      </Button>
+    )
+  }
+
+  function slettKnapp() {
+    return (
+      <Button
+        size="small"
+        variant="secondary"
+        icon={<TrashIcon aria-hidden />}
+        loading={isPending(lagreBeregningsgrunnlagResult)}
+        onClick={slettBeregningsMetodeForAvdoed}
+      >
+        Slett
+      </Button>
```

Comment:
Mer ryddig å bare legge kanppene direkte hvor de brukes enn å flytte de i egne komponenter

Reply:
ofte synes jeg disse trestrukturene blir så store og uoversiktlige, men det er mulig at det ble litt mye oppsplitting denne gangen :)

Reply:
Sånn du har satt det opp nå er i motsetting til hvordan de fleste knappene våres settes opp. Hvis det blir for store komponenter burde de heller splittes opp i egne filer med egne komponenter